pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -96,6 +96,37 @@ def test_to_datetime_format_YYYYMMDD(self, cache):
         result = pd.to_datetime(s, format='%Y%m%d', errors='coerce',
                                 cache=cache)
         expected = Series(['20121231', '20141231', 'NaT'], dtype='M8[ns]')
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('cache', [True, False])
+    def test_to_datetime_format_YYYYMMDD_overflow(self, cache):
+        # GH 25512
+        # NaN before strings with invalid date values, errors=coerce
+        s = Series(['19801222', np.nan, '20010012', '10019999'])
```

Comment:
Could you parameterize over these `Series` inputs?

Reply:
Is it what you meant?
Thanks!